src/KafkaFlow.Serializer.SchemaRegistry.ConfluentAvro/ConfluentAvroSerializer.cs

```diff
@@ -33,42 +30,39 @@ public ConfluentAvroSerializer(IDependencyResolver resolver)
             IDependencyResolver resolver,
             AvroSerializerConfig serializerConfig)
         {
-            this.schemaRegistryClient = resolver.Resolve<ISchemaRegistryClient>() ??
-                                        throw new InvalidOperationException(
-                                            $"No schema registry configuration was found. Set it using {nameof(ClusterConfigurationBuilderExtensions.WithSchemaRegistry)} on cluster configuration");
+            this.schemaRegistryClient =
+                resolver.Resolve<ISchemaRegistryClient>() ??
+                throw new InvalidOperationException(
+                    $"No schema registry configuration was found. Set it using {nameof(ClusterConfigurationBuilderExtensions.WithSchemaRegistry)} on cluster configuration");
 
             this.serializerConfig = serializerConfig;
         }
 
         /// <inheritdoc/>
         public byte[] Serialize(object message)
         {
-            if (!(message is ISpecificRecord record))
-            {
-                throw new InvalidCastException(
-                    $"The message type {message.GetType().FullName} must implement {nameof(ISpecificRecord)} interface.");
-            }
+            var serializer = Activator.CreateInstance(
```

Comment:
I think we should store this in memory (concurrentDictionary) as you did at `ConfluentSerializerWrapper`. 
So we would avoid the cost of executing `Activator.CreateInstance` for every message.

Reply:
Yes, totally agree 😃